src/bundle/Resources/public/js/scripts/core/storage.js

```
@@ -0,0 +1,57 @@
+(function (global, doc, ibexa) {
+    class IbexaStorage {
+        constructor(config) {
+            const keySuffix = config.isUserAware ? `:${ibexa.helpers.user.getId()}` : '';
+
+            this.key = `${config.key}${keySuffix}`;
+            this.eventName = config.eventName;
+        }
+
+        stringfyData(data) {
+            try {
+                return JSON.stringify(data);
+            } catch (error) {
+                console.warn('Error stringifying data', error);
+            }
+        }
+
+        parseData(data) {
+            try {
+                return JSON.parse(data);
+            } catch (error) {
```

are you sure we want to have silent error here? seems potentially error-prone